fenix/app/src/main/java/org/mozilla/fenix/gleanplumb/MessageNotificationWorker.kt

```diff
@@ -148,21 +156,36 @@ class MessageNotificationWorker(
             val notificationConfig = featureConfig.notificationConfig
             val pollingInterval = notificationConfig.refreshInterval.toLong()
 
-            val messageWorkRequest = PeriodicWorkRequest.Builder(
-                MessageNotificationWorker::class.java,
-                pollingInterval,
-                TimeUnit.MINUTES,
-            ) // Only start polling after the given interval.
-                .setInitialDelay(pollingInterval, TimeUnit.MINUTES)
-                .build()
-
             val instanceWorkManager = WorkManager.getInstance(context)
-            instanceWorkManager.enqueueUniquePeriodicWork(
-                MESSAGE_WORK_NAME,
-                // We want to keep any existing scheduled work.
-                ExistingPeriodicWorkPolicy.KEEP,
-                messageWorkRequest,
-            )
+            if (pollingInterval > 0) {
```

Comment:
Drive by - don't we should be adding conditional checks just for tests - perhaps we could pass in the work request and work manager?